modules/siddhi-core/src/main/java/org/wso2/siddhi/core/aggregation/IncrementalDataPurging.java

```diff
@@ -0,0 +1,366 @@
+                eventChunk.add(secEvent);
+                Table table = aggregationTables.get(entry.getKey());
+                try {
+                    CompiledCondition compiledCondition = table.compileCondition(expression,
+                            matchingMetaInfoHolder(table, attribute), siddhiAppContext, variableExpressionExecutorList,
+                            tableMap, "query1");
+                    LOG.info("Purging has enable for the table: " + table.getTableDefinition().getId() + " with a" +
+                            " retention of timestamp : " + purgeTime);
+
+                    table.deleteEvents(eventChunk, compiledCondition, 1);
+                } catch (Exception e) {
+                    LOG.error("Exception occurred while deleting events from " +
+                            table.getTableDefinition().getId() + " table", e);
+                    throw new DataPurgingException("Exception occurred while deleting events from " +
+                            table.getTableDefinition().getId() + " table", e);
+                }
+            }
+        }
+    }
+
+    /**
+     * Convert user provided time in to long
+     **/
+    private Long timeToLong(String value) {
+        Pattern timeValuePattern = Pattern.compile("\\d+");
+        Pattern durationPattern = Pattern.compile("\\D+");
+        Matcher timeMatcher = timeValuePattern.matcher(value);
+        Matcher durationMatcher = durationPattern.matcher(value);
+        int timeValue;
+        TimePeriod.Duration duration;
+        if (timeMatcher.find() && durationMatcher.find()) {
+            duration = durationNormalization(durationMatcher.group(0).trim());
+            timeValue = Integer.parseInt(timeMatcher.group(0));
+            try {
+                switch (duration) {
+                    case SECONDS:
+                        return Expression.Time.sec(timeValue).value();
+                    case MINUTES:
+                        return Expression.Time.minute(timeValue).value();
+                    case HOURS:
+                        return Expression.Time.hour(timeValue).value();
+                    case DAYS:
+                        return Expression.Time.day(timeValue).value();
+                    case YEARS:
+                        return Expression.Time.year(timeValue).value();
+                    default:
+                        return Expression.Time.month(timeValue).value();
+                }
+            } catch (NoSuchDurationException e) {
+                throw new NoSuchDurationException("Duration" + timeValue + "does not exists", e);
+            }
+        } else {
+            throw new NoSuchDurationException("Provided retention parameter cannot be identified. retention period:  "
+                    + value + ".");
+        }
+    }
+
+    /**
+     * normalize user provided time durations
+     */
+    private TimePeriod.Duration durationNormalization(String value) {
```

Comment:
Use verb as the function name. e.g, "NomalizDuration"

Reply:
fixed